test/box-luatest/gh_9979_tuple_format_validate_test.lua

```diff
@@ -80,22 +95,29 @@ g.test_tuple_format_validate = function()
             },
         }
 
-        for _, case in ipairs(invalid_cases) do
-            local message = string.format(
-                    "Tuple field %d (%s) type does not match one required " ..
-                    "by operation: expected %s, got %s",
-                    case.field, case.name, case.expected, case.actual
+        for _, v in ipairs(validators) do
+            for _, case in ipairs(invalid_cases) do
+                local expected = {
+                    type = 'ClientError',
+                    code = box.error.FIELD_TYPE,
+                    message = string.format(
+                        "Tuple field %d (%s) type does not match one " ..
+                        "required by operation: expected %s, got %s",
+                        case.field, case.name, case.expected, case.actual
+                    ),
+                }
+                t.assert_error_covers(
+                    expected,
+                    v.obj.validate,
+                    v.obj,
+                    case.value,
+                    string.format(
+                        "%s-validator failed for case %s",
+                        v.name, json.encode(case.value)
+                    )
                 )
-            local code = box.error.FIELD_TYPE
-
-            local expected_error = {
-                type = 'ClientError',
-                code = code,
-                message = message,
-            }
-
-            t.assert_error_covers(expected_error, format.validate, format,
-                case.value)
+            end
         end
+
     end)
 end
```

Comment:
I've noticed that if one has no access to `space`, then `box.schema.space.format(space.id)` will fail. Hence, people without access cannot read format of a space. Unfortunately, the new feature will create a security breach since all spaces are still visible to all the users and everyone would see all the formats. I don't like it at all.

Reply:
Well, if one has access to Lua, he can do whatever he wants (`os.exit()`, for example). So that's not a security breach actually.